ThePalaceProject/library-registry-admin
# Add config to run against a remote server during development.

## Description

This adds an npm script that runs library-registry-admin locally during development, using a specified remote library-registry back-end. For example:

```
npm run dev-server -- --env=backend=https://registry-tpp-qa.palaceproject.io
```

## Motivation and Context

This allows the registry admin UI to be developed and tested without needing to install and run the registry admin server locally.

I'm doing this as a first step to adding handling for 401 responses from API calls: https://www.notion.so/lyrasis/Library-Registry-Admin-React-correctly-to-a-401-API-response-696382bd7c35466aa2574925b838e316

## How Has This Been Tested?

- Clone this repo, and run `npm install`
- Run `npm run dev-server -- --env=backend=https://registry-tpp-qa.palaceproject.io`
- In a browser, open http://localhost:8080/admin
- Log in, and verify that the expected libraries appear.
- Test various functions in the UI, and verify that they work.
- Make a visible change to the code. The app should automatically update in the browser.

## Checklist:




- [x] I have updated the documentation accordingly. Instructions added to readme.
